pkg/gameservers/controller.go

```diff
@@ -253,24 +253,26 @@ func (ext *Extensions) creationMutationHandler(review admissionv1.AdmissionRevie
 		return review, nil
 	}
 
+	ext.baseLogger.Infof("ZZZML BEFORE JSON: %s", string(obj.Raw))
 	// This is the main logic of this function
 	// the rest is really just json plumbing
+	ext.baseLogger.Infof("ZZZML BEFORE Spec.Counters: %v", gs.Spec.Counters)
+	ext.baseLogger.Infof("ZZZML BEFORE Status.Counters: %v", gs.Status.Counters)
 	gs.ApplyDefaults()
+	ext.baseLogger.Infof("ZZZML AFTER Spec.Counters: %v", gs.Spec.Counters)
+	ext.baseLogger.Infof("ZZZML AFTER Status.Counters: %v", gs.Status.Counters)
 
 	newGS, err := json.Marshal(gs)
 	if err != nil {
 		return review, errors.Wrapf(err, "error marshalling default applied GameServer %s to json", gs.ObjectMeta.Name)
 	}
+	ext.baseLogger.Infof("ZZZML AFTER JSON: %s", string(newGS))
 
-	patch, err := jsonpatch.CreatePatch(obj.Raw, newGS)
+	jsonPatch, err := jsonpatch.CreateMergePatch(obj.Raw, newGS)
```

Comment:
So the issue here I think, is that this type of patch isn't supported.

See:
https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#:~:text=When%20allowing%20a,JSON%20patch%20operations.

This creates a https://datatracker.ietf.org/doc/html/rfc7396 style merge patch, whereas AdmissionReviews can only handle https://jsonpatch.com/ types of JSON Patches.

See also: https://kubernetes.io/docs/reference/config-api/apiserver-admission.v1/#admission-k8s-io-v1-AdmissionResponse

Reply:
So I did a bit more digging, and I wondered "what does `controller-runtime` use, because they have the same problem!

https://github.com/kubernetes-sigs/controller-runtime/blob/b88ed7a3602b85b6cfd0acfe9c25033b978bdb83/go.mod#L23

Which is: https://github.com/gomodules/jsonpatch

So in theory, should be a drop in replacement without any code changes other than go.mod